lib/scope.js

```diff
@@ -86,6 +86,16 @@ SymbolDef.prototype = {
             var s = this.scope;
             if (!options.screw_ie8 && this.orig[0] instanceof AST_SymbolLambda)
                 s = s.parent_scope;
+
+            // https://github.com/mishoo/UglifyJS2/issues/1095
+            // prevent mangle name conflict while using `let` or `const`
+            if (this.orig[0] instanceof AST_SymbolBlockDeclaration) {
+                var start = s.start;
+                if (start && start.type === 'keyword' && (start.value ==='default' || start.value === 'case')) {
+                    s = s.parent_scope;
+                }
+            }
+
```

Comment:
Also, a couple of test cases involving `let` use in `case` and `default` clauses would be useful.